pkg/api/licenses.go

```diff
@@ -323,35 +308,46 @@ func CreateLicense(c *gin.Context) {
 	})
 }
 
-// UpdateLicense Update license with given shortname and create audit and changelog entries.
+// UpdateLicense Update license with given id and create audit and changelog entries.
 //
 //	@Summary		Update a license
 //	@Description	Update a license in the service
 //	@Id				UpdateLicense
 //	@Tags			Licenses
 //	@Accept			json
 //	@Produce		json
-//	@Param			shortname	path		string					true	"Shortname of the license to be updated"
-//	@Param			license		body		models.LicenseUpdateDTO	true	"Update license body (requires only the fields to be updated)"
-//	@Success		200			{object}	models.LicenseResponse	"License updated successfully"
-//	@Failure		400			{object}	models.LicenseError		"Invalid license body"
-//	@Failure		404			{object}	models.LicenseError		"License with shortname not found"
-//	@Failure		409			{object}	models.LicenseError		"License with same shortname already exists"
-//	@Failure		500			{object}	models.LicenseError		"Failed to update license"
+//	@Param			id		path		string					true	"Id of the license to be updated"
+//	@Param			license	body		models.LicenseUpdateDTO	true	"Update license body (requires only the fields to be updated)"
+//	@Success		200		{object}	models.LicenseResponse	"License updated successfully"
+//	@Failure		400		{object}	models.LicenseError		"Invalid license body"
+//	@Failure		404		{object}	models.LicenseError		"License with id not found"
+//	@Failure		500		{object}	models.LicenseError		"Failed to update license"
 //	@Security		ApiKeyAuth
-//	@Router			/licenses/{shortname} [patch]
+//	@Router			/licenses/{id} [patch]
 func UpdateLicense(c *gin.Context) {
 	_ = db.DB.Transaction(func(tx *gorm.DB) error {
 		var updates models.LicenseUpdateDTO
 		var oldLicense models.LicenseDB
 
-		userId := c.MustGet("userId").(int64)
+		userId := c.MustGet("userId").(uuid.UUID)
 
-		shortname := c.Param("shortname")
-		if err := tx.Where(models.LicenseDB{Shortname: &shortname}).First(&oldLicense).Error; err != nil {
+		queryParam := c.Param("id")
```

Comment:
we should keep these names of the variables static? Like in other endpoints this same `c.Param('id')` is called `id`. From the error its apparent that its licenseId, so should we keep it that way? `licenseId`, `obligationId` etc etc?